integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateUtil.scala

```diff
@@ -493,4 +492,102 @@ object PreAggregateUtil {
     updatedPlan
   }
 
+  /**
+   * Below method will be used to get the select query when rollup policy is
+   * applied in case of timeseries table
+   * @param tableSchema
+   *                    main data map schema
+   * @param selectedDataMapSchema
+   *                              selected data map schema for rollup
+   * @return select query based on rolloup
+   */
+  def createTimeseriesSelectQueryForRollup(
```

Comment:
After compaction PR is merged , this duplicate function should be merged to there.